homeassistant/components/here_travel_time/sensor.py

```diff
@@ -0,0 +1,468 @@
+        entity = self._hass.states.get(entity_id)
+
+        if entity is None:
+            _LOGGER.error("Unable to find entity %s", entity_id)
+            return None
+
+        # Check if the entity has location attributes
+        if location.has_location(entity):
+            return self._get_location_from_attributes(entity)
+
+        # Check if device is in a zone
+        zone_entity = self._hass.states.get("zone.{}".format(entity.state))
+        if location.has_location(zone_entity):
+            _LOGGER.debug(
+                "%s is in %s, getting zone location", entity_id, zone_entity.entity_id
+            )
+            return self._get_location_from_attributes(zone_entity)
+
+        # If zone was not found in state then use the state as the location
+        if entity_id.startswith("sensor."):
+            return entity.state
+
+    @staticmethod
+    def _get_location_from_attributes(entity: State) -> str:
+        """Get the lat/long string from an entities attributes."""
+        attr = entity.attributes
+        return "{},{}".format(attr.get(ATTR_LATITUDE), attr.get(ATTR_LONGITUDE))
+
+
+class HERETravelTimeData:
+    """HERETravelTime data object."""
+
+    def __init__(
+        self,
+        origin: None,
+        destination: None,
+        app_id: str,
+        app_code: str,
+        travel_mode: str,
+        traffic_mode: bool,
+        route_mode: str,
+        units: str,
+    ) -> None:
+        """Initialize herepy."""
+        self.origin = origin
+        self.destination = destination
+        self.travel_mode = travel_mode
+        self.traffic_mode = traffic_mode
+        self.route_mode = route_mode
+        self.attribution = None
+        self.traffic_time = None
+        self.distance = None
+        self.route = None
+        self.base_time = None
+        self.origin_name = None
+        self.destination_name = None
+        self.units = units
+        self._client = herepy.RoutingApi(app_id, app_code)
+
+    def update(self) -> None:
+        """Get the latest data from HERE."""
+        if self.traffic_mode:
+            traffic_mode = TRAFFIC_MODE_ENABLED
+        else:
+            traffic_mode = TRAFFIC_MODE_DISABLED
+
+        if self.destination is not None and self.origin is not None:
+            # Convert location to HERE friendly location
+            destination = self.destination.split(",")
+            origin = self.origin.split(",")
+
+            _LOGGER.debug(
+                "Requesting route for origin: %s, destination: %s, route_mode: %s, mode: %s, traffic_mode: %s",
+                origin,
+                destination,
+                self.route_mode,
+                self.travel_mode,
+                traffic_mode,
+            )
+            response = self._client.car_route(
+                origin, destination, [self.route_mode, self.travel_mode, traffic_mode]
+            )
+            if isinstance(response, herepy.error.HEREError):
```

Comment:
Why does the library return an Exception instead of raising it?

Reply:
Because the author implemented it this way =( I can create a PR for the lib but I am not sure it will be accepted as it would alter the whole coding style of the author....

Reply:
PR in the lib was merged.